shared/api/storage_pool.go

```diff
@@ -87,7 +87,12 @@ type StoragePoolPut struct {
 // Writable converts a full StoragePool struct into a StoragePoolPut struct
 // (filters read-only fields).
 func (storagePool *StoragePool) Writable() StoragePoolPut {
-	return storagePool.StoragePoolPut
+	writable := storagePool.StoragePoolPut
```

Comment:
The other `Writable()` functions don't do this and I think its not really the place of this function to modify the struct, but rather return part of it for (potential) modification by the caller. As such I think the correct place to fix this is in the caller itself.

Great catch though thanks!

Reply:
Okey doke I'll fix in the caller :+1:

Reply:
Done.